ambry-server/src/main/java/com.github.ambry.server/AmbryRequests.java

```diff
@@ -393,16 +395,17 @@ public void handleDeleteRequest(Request request) throws IOException, Interrupted
     long startTime = SystemTime.getInstance().milliseconds();
     DeleteResponse response = null;
     try {
+      StoreKey convertedStoreKey = getConvertedStoreKeys(Collections.singletonList(deleteRequest.getBlobId())).get(0);
       ServerErrorCode error =
           validateRequest(deleteRequest.getBlobId().getPartition(), RequestOrResponseType.DeleteRequest, false);
       if (error != ServerErrorCode.No_Error) {
         logger.error("Validating delete request failed with error {} for request {}", error, deleteRequest);
         response = new DeleteResponse(deleteRequest.getCorrelationId(), deleteRequest.getClientId(), error);
       } else {
         MessageFormatInputStream stream =
-            new DeleteMessageFormatInputStream(deleteRequest.getBlobId(), deleteRequest.getAccountId(),
+            new DeleteMessageFormatInputStream(convertedStoreKey, deleteRequest.getAccountId(),
                 deleteRequest.getContainerId(), deleteRequest.getDeletionTimeInMs());
-        MessageInfo info = new MessageInfo(deleteRequest.getBlobId(), stream.getSize(), deleteRequest.getAccountId(),
+        MessageInfo info = new MessageInfo(convertedStoreKey, stream.getSize(), deleteRequest.getAccountId(),
```

Comment:
i think the notification system below also needs to use the converted ID

Reply:
Or should we note down both? 
I though we/customer may be interested in the original key's event.

Reply:
Good point - I think it should be the converted key but let's chat.

Reply:
@dharju What's your opinion about the key for notification?

Reply:
Where does this notification go exactly?  Does it go to the client ultimately?